LinqToRest/ResourceRetriever.cs

```diff
@@ -106,9 +108,42 @@ private static Uri GetResourceUri(JToken token, Uri root)
         private object GetDeserializedParameter(ParameterInfo parameter, JToken token, Uri uri)
         {
             var type = parameter.ParameterType;
+
             return type.IsQueryable()
                 ? _queryableFactory(type.GetInnerType(), uri)
-                : GetField(type, token, parameter.Name);
+                : type.IsEnum
+                    ? GetEnumFromType(type, token, parameter.Name)
+                    : GetField(type, token, parameter.Name);
+        }
+
+        private static object GetEnumFromType(Type type, JToken token, string name)
+        {
+            var getEnum = typeof(ResourceRetriever).GetMethod(nameof(GetEnum))?.MakeGenericMethod(type);
+
+            try
+            {
+                return getEnum is null
+                    ? throw new InvalidOperationException()
+                    : getEnum.Invoke(null, new object[] {token, name});
+            }
+            catch (System.Reflection.TargetInvocationException e)
+            {
+                if (e.InnerException != null)
+                {
+                    throw e.InnerException;
+                }
+                throw;
+            }
+        }
+
+        public static T GetEnum<T>(JToken token, string name) where T : struct
+        {
+            var candidate = GetField(typeof(string), token, name) as string;
+            var parsed = candidate.TryParseEnum<T>();
+
+            return parsed.Match(false, e => true)

```

Comment:
Could you add named parameters to all `Match` calls? Otherwise a reader has no clue, what's going on.

Reply:
Addressed in https://github.com/messerli-informatik-ag/linq-to-rest/commit/5757c246c8cdb57e3e94741879f2093b9801d988

Reply:
These are still not named parameters. Write `none:` and `some:` in front of the params.

Reply:
Sorry I messinterpreted your concern, now addressed in https://github.com/messerli-informatik-ag/linq-to-rest/pull/15/commits/9ceb4cf8b2ecda77a249c53bea2ec6ae6951cd0b